### PR TITLE
libjxl: update to 0.7.0

### PR DESCRIPTION
**Pull request by kmilos**
Resolves https://github.com/msys2/MINGW-packages/issues/13204

Deps bump to follow once this is reviewed.

### CODE REVIEW

File: mingw-w64-libjxl/PKGBUILD [hunk 1]
```diff
@@ -87,25 +58,26 @@ build() {
...
     "${_extra_config[@]}" \
     -DJPEGXL_ENABLE_PLUGINS=ON \
+    -DJPEGXL_ENABLE_PLUGIN_GIMP210=OFF \
+    -DJPEGXL_ENABLE_PLUGIN_MIME=OFF \
     -DBUILD_TESTING=OFF \
     -DJPEGXL_WARNINGS_AS_ERRORS=OFF \
     -DJPEGXL_ENABLE_BENCHMARK=OFF \
     -DJPEGXL_ENABLE_EXAMPLES=OFF \
     -DJPEGXL_ENABLE_MANPAGES=OFF \
+    -DJPEGXL_ENABLE_SKCMS=OFF \
```
Comment (novomesk) [inline mingw-w64-libjxl/PKGBUILD:74]: `pixbufloader-jxl.c` uses SKCMS internally so we will not avoid building it anyway.

What about using `-DJPEGXL_ENABLE_SKCMS=ON` so we don't have mixture of lcms2/skcms. What do you think?

Comment (kmilos) [reply mingw-w64-libjxl/PKGBUILD:74]: I prefer the plugin implemented the choice so we can go all lcms2 ;)

**Comment by novomesk** [comment]
I think `-DJPEGXL_ENABLE_SJPEG=OFF` helps to avoid sjpeg completely.

File: mingw-w64-libjxl/PKGBUILD (hunk 1 above)
Comment (novomesk) [reply mingw-w64-libjxl/PKGBUILD:74]: For me doesn't matter too much when both choices work correctly.

**Comment by kmilos** [comment]
I'm also 50/50 about (not) shipping the MIME info... Although present in `shared-mime-info`, I guess there are non-GTK apps that could still benefit from this? We ship it in `libavif` and `libheif`...

**Comment by novomesk** [comment]
Without shared-mime-info it doesn't make sense. Duplicate definition is not necessary.

**Comment by kmilos** [comment]
> Without shared-mime-info it doesn't make sense. Duplicate definition is not necessary.

TODO then to remove from `libavif` and `libheif`...

**Comment by novomesk** [comment]
I know libheif adds their own XML files but libavif doesn't.

**Comment by kmilos** [comment]
I also tried enabling the man pages, but still no dice... Reported upstream though: https://github.com/libjxl/libjxl/issues/1772

**Comment by novomesk** [comment]
Until resolved, we can continue to live without manpages.

**Comment by Biswa96** [comment]
Wait, found an workaround. Would you like to try this patch?

```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -442,8 +442,8 @@ if (ASCIIDOC_PY_FOUND)
     # does not recognize it.
     add_custom_command(
       OUTPUT "${PAGE}.1"
-      COMMAND "${ASCIIDOC_PY}"
-      ARGS ${ASCIIDOC}
+      COMMAND ${ASCIIDOC}
+      ARGS
         --format manpage --destination-dir="${CMAKE_CURRENT_BINARY_DIR}"
         "${CMAKE_CURRENT_SOURCE_DIR}/doc/man/${PAGE}.txt"
       MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/doc/man/${PAGE}.txt")
```

Make sure to enable `JPEGXL_ENABLE_MANPAGES` option in cmake. All credit goes to https://stackoverflow.com/a/59635064

**Comment by kmilos** [comment]
> Wait, found an workaround.

Yeah, I tried something similar (equivalent setting on line 410), and reported in the upstream issue, thanks.

**Comment by kmilos** [comment]
Next, this new `cjxl_hdr` looks pretty large, and seems statically linked, don't know why...

**Comment by novomesk** [comment]
When I tried older snapshot I got 93.7 MB for cjpeg_hdr.exe

On my Gentoo Linux I have only:
`-rwxr-xr-x 1 root root 2724016 sep 14 19:08 /usr/bin/cjpeg_hdr`

**Comment by kmilos** [comment]
Yeah, it's around 3 MB here as well... (about the same as the libjxl library itself)

I don't see any linking in tools/CMakeLists.txt, it really does look like it is source only.

So I think this is good to go now unless there are other objections.

The grokker didn't complain, does it really mean no bumping of deps is necessary?

**Comment by novomesk** [comment]
I wish that next-gen `gimp3` and `kimageformats-qt5` are rebuilt.

And that libjxl is added as dependency for old 2.10.32 `gimp`

**Comment by kmilos** [comment]
Any other feedback? Shall I bump the dependants next?

**Comment by Biswa96** [comment]
Would you like to rebase the changes to latest commit of this repository? The kimageformats-qt5 looks different, no clangarm64.

**Comment by kmilos** [comment]
Great... I guess Clang 15 is now more strict about even more stuff:

```
 ../../gimp-2.10.32/libgimp/gimpprogressbar.c:192:12: error: incompatible pointer to integer conversion returning 'HGDIOBJ' (aka 'void *') from a function with result type 'guint32' (aka 'unsigned int') [-Wint-conversion]
      return GDK_WINDOW_HWND (gtk_widget_get_window (GTK_WIDGET (window)));
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

@novomesk Can you please provide a patch for Gimp 2.10.x?

**Comment by Biswa96** [comment]
Workaround `CFLAGS+=" -Wno-int-conversion" CXXFLAGS+=" -Wno-int-conversion"`

**Comment by kmilos** [comment]
Looks like it's the same for Gimp 2.99.x 😞

**Comment by kmilos** [comment]
At least the Gimp build seems to be passing now... That'll be fun for the arm runner 😮

**Comment by MehdiChinoune** [comment]
Shouldn't this PR only for updating jxl, Enabling jxl should be done separately in other PRs.

**Comment by kmilos** [comment]
I'll split this indeed, the build times are very long...

**Comment by MehdiChinoune** [comment]
It looks like python-imagecodecs should be patched to build against blosc2>=2.3.0 https://github.com/msys2/MINGW-packages/actions/runs/2924965192/jobs/4665294829

**Comment by MehdiChinoune** [comment]
If someone familiar with cython could add `#define BLOSC1_COMPAT 1` to https://github.com/cgohlke/imagecodecs/blob/master/imagecodecs/_blosc2.pyx

**Comment by kmilos** [comment]
I'll probably just turn off that error temporarily as well (a similar one fails GraphicsMagick too).

Let's see how upstream patches iz: https://github.com/cgohlke/imagecodecs/issues/47

**Comment by MehdiChinoune** [comment]
I am going to patch blosc2 instead

**Comment by kmilos** [comment]
Let's be pragmatic and just turn the error of until the upstream fix.

Btw, look like a lot of projects are being bitten by these two issues since Clang 15 came out:

https://discourse.llvm.org/t/configure-script-breakage-with-the-new-werror-implicit-function-declaration/65213/3

**Comment by kmilos** [comment]
It further looks like these were downgraded to warnings in 15.0.1 and I rebased this after @lazka updated it, but `python-imagecodecs` must use `-Wall` somewhere...

**Comment by MehdiChinoune** [comment]
building python-imagecodecs should be fixed now.

**Comment by kmilos** [comment]
Thanks!

**Comment by kmilos** [comment]
We might have to disable manpages on arm64, a2x/xmllint fails there for some reason...

**Comment by Biswa96** [comment]
libjxl build was failed due to error in path relocation of libxml2.

**Comment by kmilos** [comment]
Great, I saw the libxml2 update and the successful build now, thanks all!

**Pull request merged**